src/map/skill.cpp

```diff
@@ -3458,6 +3458,14 @@ int64 skill_attack (int attack_type, struct block_list* src, struct block_list *
 
 	damage = dmg.damage + dmg.damage2;
 
+	if (tsc && tsc->data[SC_MAXPAIN]) {
+		tsc->data[SC_MAXPAIN]->val3 = (int)damage;
+		tsc->data[SC_MAXPAIN]->val2 = 0;
+		if (!tsc->data[SC_KYOMU]) {//SC_KYOMU invalidates reflecting ability
```

Comment:
Should SC_DARKCROW short range also be checked here like you did in battle?

Reply:
This part is for magic damage I thought, right?

Darkcrow has no effect to magic reflecting ability.

Or should I add Magic attack trigger on code?

Actually when I tested it works well though.

Reply:
It was purely a question, I'm not familiar with this part of Max Pain, but after doing some reading on Dark Crow it sounds like what you have should be good!

Reply:
Nice!